hw/ip/otbn/util/otbn_sim_test.py

```diff
@@ -87,39 +73,68 @@ def main() -> int:
                         help=('File containing expected dmem values. '
                               'Addresses that are not listed are allowed to '
                               'have any value.'))
+    parser.add_argument('--testcase',
+                        metavar='FILE',
+                        type=argparse.FileType('r'),
+                        help='Path to a testcase hjson file.')
     parser.add_argument('elf',
                         help='Path to the .elf file for the OTBN program.')
     parser.add_argument('-v', '--verbose', action='store_true')
     args = parser.parse_args()
 
+    _, _, symbols = read_elf(args.elf)
+
     # Parse expected values.
     result = CheckResult()
 
+    cmd_flags = []
+
+    testcase = None
+    if args.testcase:
```

Comment:
I think we need to check somewhere around here that we don't have e.g. both `--expected_dmem` and `--testcase`. At the moment, I think `--testcase` is winning, but that's probably a bit confusing :-)